src/lib/components/Radio/Radio.jsx

```diff
@@ -36,53 +37,59 @@ export const Radio = ({
         required && styles.isRootRequired,
         getRootValidationStateClassName(validationState, styles),
       )}
+      disabled={disabled}
       id={id}
     >
+      <legend
+        className={styles.legend}
+        id={id && `${id}__label`}
+      >
+        {label}
+      </legend>
       <div
+        aria-hidden
         className={classNames(
           styles.label,
           !isLabelVisible && styles.isLabelHidden,
         )}
-        id={id && `${id}__labelText`}
+        id={id && `${id}__displayLabel`}
       >
         {label}
       </div>
       <div className={styles.field}>
-        <ul className={styles.list}>
```

Comment:
Why are we dropping the list based implementation?
Isn't list more semantic? 

Let me know, I'm eager for knowledge :books:

Reply:
[My idea is that] Once there is the `<fieldset>` element, the inputs are already tied together which makes the `<ul>` redundant.

Reply:
(Also, Heydon [does it](https://www.sitepoint.com/replacing-radio-buttons-without-replacing-radio-buttons/) the same way.)